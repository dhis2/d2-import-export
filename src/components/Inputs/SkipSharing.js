import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'
import { Field } from '../Field/Field'
import { Label } from '../Field/Label'

export const OPTION_YES = { value: 'true', label: i18n.t('Yes') }
export const OPTION_NO = { value: 'false', label: i18n.t('No') }
export const SKIP_SHARING_KEY = 'skipSharing'
export const SKIP_SHARING_DEFAULT_VALUE = OPTION_NO.value

export const SkipSharing = () => (
    <Field>
        <Label>{i18n.t('Skip sharing')}</Label>
        <RadioGroup
            name={SKIP_SHARING_KEY}
            label={i18n.t('Skip sharing')}
            options={[OPTION_YES, OPTION_NO]}
            dataTest="input-skip-sharing"
        />
    </Field>
)
