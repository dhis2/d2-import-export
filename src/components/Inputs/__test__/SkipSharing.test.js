import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { SkipSharing } from '../SkipSharing'

jest.mock('react-final-form')

describe('Input component - SkipSharing', () => {
    it('should render correctly', () => {
        const file = shallow(<SkipSharing />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
