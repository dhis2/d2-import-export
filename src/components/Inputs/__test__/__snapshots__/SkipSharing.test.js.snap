// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - SkipSharing should render correctly 1`] = `
<RadioGroup
  dataTest="input-skip-sharing"
  label="Skip sharing"
  name="skipSharing"
  options={
    Array [
      Object {
        "label": "Yes",
        "value": "true",
      },
      Object {
        "label": "No",
        "value": "false",
      },
    ]
  }
/>
`;
